mgmtfn/dockcontivnet/main.go

```diff
@@ -0,0 +1,474 @@
+		//       "HostPort": 1234
+		//     }
+		//
+
+		logEvent("create endpoint")
+
+		content, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			httpError(w, "Could not read endpoint create request", err)
+			return
+		}
+
+		cereq := api.CreateEndpointRequest{}
+
+		if err := json.Unmarshal(content, &cereq); err != nil {
+			httpError(w, "Could not read endpoint create request", err)
+			return
+		}
+
+		if err := netdcliAdd(generateEndpoint(cereq.EndpointID, tenantName, networkName, hostname)); err != nil {
+			httpError(w, "Could not create endpoint", err)
+			return
+		}
+
+		ipaddr, err := netdcliGet("ipAddress", "endpoint", networkName+"-"+cereq.EndpointID)
+		if err != nil {
+			httpError(w, "Could not find created endpoint", err)
+			return
+		}
+
+		subnetLen, err := netdcliGet("subnetLen", "network", networkName)
+		if err != nil {
+			httpError(w, "Could not find subnet length in network definition", err)
+			return
+		}
+
+		log.Debug(ipaddr)
+
+		epResponse := api.CreateEndpointResponse{
+			Interfaces: []*api.EndpointInterface{
+				&api.EndpointInterface{
+					ID:      1,
+					Address: ipaddr + "/" + subnetLen,
+				},
+			},
+		}
+
+		content, err = json.Marshal(epResponse)
+		if err != nil {
+			httpError(w, "Could not generate create endpoint response", err)
+			return
+		}
+
+		w.Write(content)
+	}
+}
+
+func endpointInfo(w http.ResponseWriter, r *http.Request) {
+	logEvent("endpoint info")
+
+	content, err := json.Marshal(api.EndpointInfoResponse{})
+	if err != nil {
+		httpError(w, "Could not generate endpoint info response", err)
+		return
+	}
+
+	w.Write(content)
+}
+
+func join(networkName string) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		logEvent("join")
+
+		jr := api.JoinRequest{}
+		content, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			httpError(w, "Could not read join request", err)
+			return
+		}
+
+		if err := json.Unmarshal(content, &jr); err != nil {
+			httpError(w, "Could not parse join request", err)
+			return
+		}
+
+		ifName, err := netdcliGet("portName", "endpoint", networkName+"-"+jr.EndpointID)
+		if err != nil {
+			httpError(w, "Could not derive created interface", err)
+			return
+		}
+
+		defaultGw, err := netdcliGet("defaultGw", "network", networkName)
+		if err != nil {
+			httpError(w, "Could not retrieve DefaultGw", err)
+			return
+		}
+
+		content, err = json.Marshal(api.JoinResponse{
+			InterfaceNames: []*api.InterfaceName{
+				&api.InterfaceName{},
+				&api.InterfaceName{
+					SrcName: ifName,
+					DstName: "eth",
+				},
+			},
+			Gateway: defaultGw,
+		})
+
+		if err != nil {
+			httpError(w, "Could not generate join response", err)
+			return
+		}
+
+		w.Write(content)
+	}
+}
+
+func leave(networkName string) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		logEvent("leave")
+		w.WriteHeader(200)
+	}
+}
+
+func netdcli(payload string, add bool) error {
```

Comment:
the `netdcli` related function can be implemented by using the methods provided by `netmaster/client.Client`. This shall eliminate the need for spawning the netdcli binary and screen scrapping.